### PR TITLE
Remove usage of md5 for checksums in Agent dependencies

### PR DESCRIPTION
**Pull request by albertvaka**
Use sha256 checksums in software definitions used by the Agent.

Omnibus won't accept md5 nor sha1 anymore after we merge https://github.com/DataDog/omnibus-ruby/pull/136

Some definitions we don't use still have md5 and sha1 hashes, but that's
fine because if we ever try to use them as-is, omnibus will raise an error.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

**Comment by cedricvanrompay-datadog** [comment]
Hi, how were the SHA256 values obtained?

File: config/software/makedepend.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
 default_version "1.0.5"
 
 source url: "http://xorg.freedesktop.org/releases/individual/util/makedepend-1.0.5.tar.gz",
```
Comment (cedricvanrompay-datadog) [inline config/software/makedepend.rb:21]: Would it be possible to switch to HTTPS?

(several other occurrences, I'd recommend doing a global search across  the repo for `http` without a `s`)

Comment (albertvaka) [reply config/software/makedepend.rb:21]:
```suggestion
source url: "https://xorg.freedesktop.org/releases/individual/util/makedepend-1.0.5.tar.gz",
```

File: config/software/util-macros.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 default_version "1.19.0"
 
 source url: "http://xorg.freedesktop.org/releases/individual/util/util-macros-#{version}.tar.gz",
```
Comment (albertvaka) [inline config/software/util-macros.rb:5]:
```suggestion
source url: "https://xorg.freedesktop.org/releases/individual/util/util-macros-#{version}.tar.gz",
```

**Comment by albertvaka** [comment]
@cedricvanrompay-datadog The SHA256 values were calculated on my laptop. The upstream projects either don't provide checksums or only provide MD5. If for some reasons we got different hashes on my laptop and on the CI, the build would fail.

File: config/software/xproto.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 default_version "7.0.27"
 
 source url: "http://xorg.freedesktop.org/releases/individual/proto/xproto-#{version}.tar.gz",
```
Comment (albertvaka) [inline config/software/xproto.rb:5]:
```suggestion
source url: "https://xorg.freedesktop.org/releases/individual/proto/xproto-#{version}.tar.gz",
```

File: config/software/libiconv.rb [hunk 4]
```diff
@@ -22,7 +22,7 @@
 dependency "config_guess"
 
 source url: "http://ftp.gnu.org/pub/gnu/libiconv/libiconv-#{version}.tar.gz",
```
Comment (albertvaka) [inline config/software/libiconv.rb:24]:
```suggestion
source url: "https://ftp.gnu.org/pub/gnu/libiconv/libiconv-#{version}.tar.gz",
```

File: config/software/liblzma.rb [hunk 5]
```diff
@@ -19,7 +19,7 @@
 default_version "5.0.5"
 
 source url: "http://tukaani.org/xz/xz-#{version}.tar.gz",
```
Comment (albertvaka) [inline config/software/liblzma.rb:21]:
```suggestion
source url: "https://tukaani.org/xz/xz-#{version}.tar.gz",
```

File: config/software/libyaml.rb [hunk 6]
```diff
@@ -30,7 +30,7 @@
 
 build do
   license "MIT"
-  license_file "./LICENSE"
+  license_file "https://raw.githubusercontent.com/yaml/libyaml/#{version}/LICENSE"
```
Comment (KSerrania) [inline config/software/libyaml.rb:33]: Given that we do check the tar.gz sum but not the license sum, I'd argue that using the license file provided in the tar file is safer (and that's one less download). Was there a specific reason to stop using the file provided in the tar.gz?

Comment (albertvaka) [reply config/software/libyaml.rb:33]:
```suggestion
  license_file "./LICENSE"
```

**Pull request merged**